### PR TITLE
useScrollStates hook created replacing direct access to scrollTopState & scrollLeftState

### PR DESCRIPTION
**Pull request by Jeetch8**
Hii , I have made the changes. Can you please check and give me a feedback on that ?
As I am a bit uncertain on it.
here is the issue #4826 
Thank u

### CODE REVIEW

File: packages/twenty-front/src/modules/ui/utilities/scroll/hooks/useScrollSates.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { useAvailableScopeIdOrThrow } from "@/ui/utilities/recoil-scope/scopes-internal/hooks/useAvailableScopeId";
+import { scrollLeftComponentState } from "@/ui/utilities/scroll/states/scrollLeftComponentState";
+import { scrollTopComponentState } from "@/ui/utilities/scroll/states/scrollTopComponentState";
+import { extractComponentState } from "@/ui/utilities/state/component-state/utils/extractComponentState"
+import { ViewScopeInternalContext } from "@/views/scopes/scope-internal-context/ViewScopeInternalContext";
+
+export const useScrollStates = (scrollWrapperId?:string) => {
```
[the remark this answers is not in the document]
Comment (charlesBochet) [reply packages/twenty-front/src/modules/ui/utilities/scroll/hooks/useScrollSates.ts:7]: could you move this hook into a internal subfolder within hooks. internal hooks cannot be used outside of their component (here Scroll / ScrollWrapper).
This allows us to narrow down the public API of a component to enforce interface (easy maintenance).

We will need to create another useScrollTopValue(scrollWraperId) and useScrollLeftValue(scrollWraperId) hooks. These hooks will only expose what functions we want it to expose.

Here we would for example only want to expose the getter: useRecoilValue(scrollLeftState) (scrollLeftState being imported from useScrollStates)

Then from ColumnHead, Recordtable, RecordTableHeaderCell you can use useScrollTopValue and useScrollLeftValue

**Review by charlesBochet — COMMENTED** [review]
Hi @Jeetch8, thank you for your work! I have left some guidance on how to use hooks and component States. The doc is inexistant about it for now but I can guide you more if needed

**Comment by charlesBochet** [comment]
Also, make sure to run yarn nx lint:ci twenty-front before pushing!

**Comment by FelixMalfait** [comment]
Hey, I'm sorry I will close this since it hasn't move since the last comments a month ago.

**Pull request closed without merging**